Lobby/EOSLobby.cs

```diff
@@ -0,0 +1,197 @@
+
+            //set the mod handle
+            EOSSDKComponent.GetLobbyInterface().UpdateLobbyModification(new UpdateLobbyModificationOptions 
+            { LobbyId = callback.LobbyId, LocalUserId = EOSSDKComponent.LocalUserProductId }, out modHandle);
+
+            //add attributes
+            modHandle.AddAttribute(new LobbyModificationAddAttributeOptions { Attribute = defaultData, Visibility = LobbyAttributeVisibility.Public });
+            modHandle.AddAttribute(new LobbyModificationAddAttributeOptions { Attribute = hostAddressData, Visibility = LobbyAttributeVisibility.Public });
+
+            //add user attributes
+            if (lobbyData != null)
+            {
+                foreach (AttributeData data in lobbyData)
+                {
+                    modHandle.AddAttribute(new LobbyModificationAddAttributeOptions { Attribute = data, Visibility = LobbyAttributeVisibility.Public });
+                    lobbyReturnData.Add(new Attribute { Data = data, Visibility = LobbyAttributeVisibility.Public });
+                }
+            }
+
+            //update the lobby
+            EOSSDKComponent.GetLobbyInterface().UpdateLobby(new UpdateLobbyOptions { LobbyModificationHandle = modHandle }, null, (UpdateLobbyCallbackInfo _) => { });
+
+            isLobbyOwner = true;
+            ConnectedToLobby = true;
+            EOSSDKComponent.GetLobbyInterface().CopyLobbyDetailsHandle(new CopyLobbyDetailsHandleOptions { LobbyId = callback.LobbyId, LocalUserId = EOSSDKComponent.LocalUserProductId }, out ConnectedLobbyDetails);
+            currentLobbyId = callback.LobbyId;
+
+            //invoke event
+            JoinLobbyComplete.Invoke(lobbyReturnData);
+        });
+    }
+
+    //find lobbies
+    public virtual void FindLobbies(uint maxResults, LobbySearchSetParameterOptions[] lobbySearchSetParameterOptions)
+    {
+        //create search handle and list of lobby details
+        LobbySearch search = new LobbySearch();
+
+        //set the search handle
+        EOSSDKComponent.GetLobbyInterface().CreateLobbySearch(new CreateLobbySearchOptions { MaxResults =  maxResults}, out search);
+
+        //set search parameters
+        foreach (LobbySearchSetParameterOptions searchOption in lobbySearchSetParameterOptions)
+        {
+            search.SetParameter(searchOption);
+        }
+        
+        //find lobbies
+        search.Find(new LobbySearchFindOptions { LocalUserId = EOSSDKComponent.LocalUserProductId }, null, (LobbySearchFindCallbackInfo callback) => 
+        {
+            List<LobbyDetails> foundLobbies = new List<LobbyDetails>();
+
+            //if the search was unsuccessful, log an error and return
+            if (callback.ResultCode != Result.Success)
+            {
+                Debug.LogError("There was an error while finding lobbies. Error: " + callback.ResultCode);
```

Comment:
Also have an event invoke here in case some ui is waiting for an event to finish a loading screen etc.